sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -414,13 +407,12 @@ abstract class BaseHashJoinIterator(
     // Based off of the keys on the build side guess at how many output rows there
     // will be for each input row on the stream side. This does not take into account
     // the join type, data skew or even if the keys actually match.
-    withResource(countGroups(builtKeys)) { builtCount =>
-      val isDistinct = builtCount.getRowCount == builtKeys.numRows()
-      val counts = builtCount.getColumn(builtCount.getNumberOfColumns - 1)
-      withResource(counts.reduce(ReductionAggregation.mean(), DType.FLOAT64)) { scalarAverage =>
-        (scalarAverage.getDouble, isDistinct)
-      }
+    val builtCount = withResource(GpuColumnVector.from(builtKeys)) { keysTable =>
+      keysTable.distinctCount(NullEquality.EQUAL)
```

Comment:
with `NullEquality.EQUAL` would we undercount in comparison to `NullPolicy.INCLUDE`?

Reply:
We need NullEquality.EQUAL here because there can be struct join keys that have embedded nulls.  This also matches the treatment of top-level nulls in the original groupby, as those will be grouped into a single value as well.